### ISSUE TITLE
Copy from Cache To Store Strips Content Type

### ISSUE THREAD

**Issue by curreta** [renchap/shrine-google_cloud_storage#36]
The original image is successfully being stored in the 'cache' storage, and subsequently when it comes time to upload, it makes it successfully to the ```copy``` [method](https://github.com/renchap/shrine-google_cloud_storage/blob/master/lib/shrine/storage/google_cloud_storage.rb#L29)

However, the original file that gets copied over to the main 'store' storage will lose it's Content Type. 

I tested using the copy functionality directly using the google-cloud-storage gem. The content type remains.

I have been unable to track down the issue.

![Screenshot_2019-08-03 Bucket details - ResidentHaven - Google Cloud Platform](https://user-images.githubusercontent.com/284943/62418126-fad8c200-b627-11e9-8823-f7cec3723f62.png)

**Update 1**
I was able to narrow it down by deleting [this do loop](https://github.com/renchap/shrine-google_cloud_storage/blob/master/lib/shrine/storage/google_cloud_storage.rb#L33) - It appears those options are overwriting the content type.

**Update 2**
Setting any custom headers as described on the readme will cause the content-type to be emptied

```
Shrine::Storage::GoogleCloudStorage.new(
  bucket: "store",
  default_acl: 'publicRead',
  object_options: {
    cache_control: 'public, max-age: 7200'
  },
)
```

**Comment by renchap** [comment]
Thanks for this report. I am not sure why setting those additional attributes removes the content type, I will try to troubleshoot this.

**Comment by renchap** [comment]
This is a bug in the `google-cloud-storage` gem. I filled a bug on their repo: https://github.com/googleapis/google-cloud-ruby/issues/4254

**Comment by renchap** [comment]
I just released 2.0.1 with a workaround for this issue, until this is fixed upstream.

**Comment by quartzmo** [comment]
I have opened PR googleapis/google-cloud-ruby#4275 to fix this issue in general in `google-cloud-storage`. Please take a look, comments welcome.

**Comment by quartzmo** [comment]
Fix released in Released in [google-cloud-storage v1.24.0](https://rubygems.org/gems/google-cloud-storage/versions/1.24.0).
